server/src/main/java/io/crate/metadata/doc/DocTableInfoFactory.java

```diff
@@ -201,6 +205,11 @@ public void validateSchema(IndexMetadata indexMetadata) {
     }
 
     public DocTableInfo create(RelationName relation, Metadata metadata) {
+        RelationMetadata relationMetadata = metadata.getRelation(relation);
```

Comment:
Wouldn't it face the same BWC on upgrade issue as blob did in https://github.com/crate/crate/pull/17345/files#diff-297915f4c37ad1828c9daa8c8a7fcab897e43ba2e478ee2ca3f2cf4ecf0af6aeL55?